ush/setup.py

```diff
@@ -1082,78 +1063,22 @@ def get_location(xcs, fmt, expt_cfg):
-                sub-hourly output is not really being performed:
-                  DT_SUBHOURLY_POST_MNTS = \"{DT_SUBHOURLY_POST_MNTS}\"
-                Resetting SUB_HOURLY_POST to \"FALSE\".  If you do not want this, you
-                must set DT_SUBHOURLY_POST_MNTS to something other than zero."""
-            )
-            post_config["SUB_HOURLY_POST"] = False
-
-        if dt_subhourly_post_mnts < 1 or dt_subhourly_post_mnts > 59:
-            raise ValueError(
-                f'''
-                When SUB_HOURLY_POST is set to \"TRUE\",
-                DT_SUBHOURLY_POST_MNTS must be set to an integer between 1 and 59,
-                inclusive but:
-                  DT_SUBHOURLY_POST_MNTS = \"{dt_subhourly_post_mnts}\"'''
-            )
-
-        # Check that DT_SUBHOURLY_POST_MNTS (after converting to seconds) is
-        # evenly divisible by the forecast model's main time step DT_ATMOS.
-        dt_atmos = fcst_config["DT_ATMOS"]
-        rem = dt_subhourly_post_mnts * 60 % dt_atmos
-        if rem != 0:
-            raise ValueError(
-                f"""
-                When SUB_HOURLY_POST is set to \"TRUE\") the post
-                processing interval in seconds must be evenly divisible
-                by the time step DT_ATMOS used in the forecast model,
-                i.e. the remainder must be zero.  In this case, it is
-                not:
-
-                  DT_SUBHOURLY_POST_MNTS = \"{dt_subhourly_post_mnts}\"
-                  DT_ATMOS = \"{dt_atmos}\"
-                  remainder = (DT_SUBHOURLY_POST_MNTS*60) %% DT_ATMOS = {rem}
-
-                Please reset DT_SUBHOURLY_POST_MNTS and/or DT_ATMOS so
-                that this remainder is zero."""
-            )
+    post_config = expt_config["task_run_post"]
 
     # Make sure the post output domain is set
-    predef_grid_name = workflow_config.get("PREDEF_GRID_NAME")
-    post_output_domain_name = post_config.get("POST_OUTPUT_DOMAIN_NAME")
+    post_output_domain_name = post_config["post_output_domain_name"]
 
-    if not post_output_domain_name:
-        if not predef_grid_name:
-            raise Exception(
-                f"""
-                The domain name used in naming the run_post output files
-                (POST_OUTPUT_DOMAIN_NAME) has not been set:
-                POST_OUTPUT_DOMAIN_NAME = \"{post_output_domain_name}\"
-                If this experiment is not using a predefined grid (i.e. if
-                PREDEF_GRID_NAME is set to a null string), POST_OUTPUT_DOMAIN_NAME
-                must be set in the configuration file (\"{user_config}\"). """
-            )
-        post_output_domain_name = predef_grid_name
+    if "{{ " in post_output_domain_name:
+        raise Exception(
+            f"""
 
-    if not isinstance(post_output_domain_name, int):
-        post_output_domain_name = lowercase(post_output_domain_name)
+            The preferred output files from post has not been set:
```

Comment:
`files` and `has` don't agree -- either `file` + `has` or `files` + `have`, if I'm not misunderstanding. Or something like `The post output-file preference has not been set`? I don't exactly understand the context so that might not be right, but I think the grammar of this message is off.

Reply:
Totally. I think it should read `The preferred output domain from post has not been set`.